modules/file-rotation/file-rotation.c

```diff
@@ -24,11 +24,40 @@
 #include "file-rotation.h"
 #include "modules/affile/file-signals.h"
 #include "driver.h"
+#include "time.h"
 
 static void
 _slot_file_rotation(FileRotationPlugin *self, FileFlushSignalData *data)
 {
   msg_trace("File rotation signal received", evt_tag_str("filename", data->filename));
+
+  if(data != NULL)
+    {
+      if(data->size >= self->size)
+        {
+          msg_debug("File size reached the limit", evt_tag_str("filename", data->filename));
+          msg_debug("Sending file rotation signal", evt_tag_str("filename", data->filename));
+
+          time_t now = time(NULL);
+          struct tm *tm = localtime(&now);
+          gchar *date = g_new0(gchar, 100);
+
+          strftime(date, 100, self->date_format, tm);
```

Comment:
- So there's a need of new option which tells how many times a file can be rotated ?

- And an option validation method inside `file_rotation_attach_to_driver()` method ?

Reply:
I'm not sure we have the time to add the `rotate()` option, but feel free to implement it whenever you have time between review rounds.

With `file-rotation(size(300M), rotate(3))`, the new filename would be originalfilename.1, older files would be originalfilename.2 and originalfilename.3. rotate(3) means that we keep only the 3 latest files and remove the older ones.

In case you can implement this, we should not make `date-format()` mandatory.

In case you don't have time to implement the above functionality, we should make `date-format()` mandatory (by adding a validation check to file_rotation_attach_to_driver()).

Reply:
Yeah I've implemented the same https://imgur.com/a/acd9KiW